docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,632 @@
+databases.
+
+#### CHAR
+
+The `CHAR(x)` type is actually a string of max length `x`. If a value you insert isn't the full length `x`, then the 
+value will be right-padded with spaces. For example, if the column type is `CHAR(5)`, then "hello" would be saved as 
+"hello", but "hey" would be saved as "hey__" (pretend the underscores are spaces".
+
+#### VARCHAR
+
+The `VARCHAR(x)` type is a variable-length string of max length `x`. If a value you insert isn't the full length `x`, 
+then the value will _not_ be right-padded with spaces. For example, if the column type is `VARCHAR(5)`, then "hello" 
+would be saved as "hello" and "hey" would be saved as "hey".
+
+
+##### CHAR vs VARCHAR
+
+The difference between `CHAR(x)` and `VARCHAR(x)` is that a `CHAR` will right-pad the entry with spaces if it isn't of 
+length `x`, but a `VARCHAR` would save the entered value. If either columns attempt to save a value that is longer than
+`x` characters, then the value will be truncated and an error or warning might be thrown.
+
+#### TEXT
+
+The `TEXT` type is a string of unlimited length. The text will not be truncated.
+
+##### VARCHAR/CHAR vs TEXT
+
+The difference between `CHAR(x)`/`VARCHAR(x)` and `TEXT` is that a `CHAR`/`VARCHAR` is limited by the length of `x`. If
+an entry is made with length greater than `x`, the value will be truncated and an error or warning might be thrown. 
+`TEXT` will not be truncated. It's generally more efficient to store and perform operations on a `CHAR` or `VARCHAR` 
+so if possible, try to use those. Columns like usernames, first/last names, emails, website URLs, and others, whose 
+length generally won't be longer than a certain length should be stored as that.
+
+#### INTEGER/BIGINTEGER
+
+The `INTEGER` or `INT` type is an integer with a range of [-2147483648, 2147483647]. The `BIGINTEGER`/`BIGINT`
+is an integer value with an extended range of [-9223372036854775808, 9223372036854775807] (wow that was terrible to type
+out). 
+
+#### DECIMAL
+
+The `DECIMAL` type is similar to a double/float in other programming languages. You can usually specify the precision
+and scale of the column by writing it like this: `DECIMAL(19, 2)`. 
+
+**Precision** defines the total count of significant digits in the number (total number of digits on *both sides* of the
+decimal point).
+**Scale** defines the count of digits to the right of the decimal.
+
+The range of this data type is *up to 131072 digits before the decimal point; up to 16383 digits after the decimal 
+point*.
+
+#### TIMESTAMP
+
+The `TIMESTAMP` type is one of the date/time types commonly seen in SQL databases. There's also a `TIMESTAMP` type with
+a time zone, if that suits your needs better. 
+
+#### BOOLEAN/TINYINTEGER
+
+The `BOOLEAN` is a type which is available in PostgreSQL. In many other DBs, `TINYINTEGER` or `TINYINT` is commonly used
+instead (they are the same thing). 
```

Comment:
same here, mention that TINYINT has values of 0=false or 1=true

Reply:
Added.